Fix unformatted dates on LiveViews

Bentheburrito opened the pull request:
This PR should fix the unformatted dates on new/updated elements, closing #140. I refactored a bit of the TS formatting code, so let me know if it looks okay.

I had some trouble using `push_event` with the `phx:...` listeners, since the TypeScript `Event` type doesn't have the `detail` property that Phoenix adds (so the compiler got mad). So I used a Hook instead

dm0896665 reviewed (approved):
For events like that, to get the `detail` from them you have to cast them to a `CustomEvent` like so:
```typescript
document.getElementById(this.tableID)?.addEventListener("phx:custom_event", (customEvent: Event) => {
        console.log((<CustomEvent>customEvent).detail[0]);
      });
```

With that being said I don't think it fully closes #140 as it doesn't handle other formats such as
- [x] Commas to numbers
- [x] Seconds to HHMMSS
- [ ] Add percents

Are you wanting just to add that in another issue?
If you are, this is working great! And can be merged in.

Bentheburrito commented:
Thank you for catching that I missed the other 3 formatting fns. I added hooks for those, though the session page only currently uses `addCommasToNumber()` and of course `formatDateTime()`.

I also found 2 small bugs while testing that are included here:
1. on the search page, if a non-PlayerLogin/PlayerLogout event was received, there'd be a `no function clause matching in SacaStatsWeb.CharacterLive.Search.handle_info/2` error, so I added a catch-all fn head to avoid that.
2. on the session page, if you were viewing an older session, you'd still receive new events. I wrapped the `PubSub.subscribe` call in an if statement to check if this is the current session to avoid that.

Let me know if this looks good!
